crates/api/src/transport.rs

```diff
@@ -101,15 +102,28 @@ impl TxImpHnd {
     ) -> BoxFut<'_, K2Result<()>> {
         Box::pin(async move {
             let data = K2Proto::decode(&data)?;
-            let ty = data.ty();
+            let message_type = data.ty();
             let K2Proto {
                 space_id,
                 module_id,
                 data,
                 ..
             } = data;
 
-            match ty {
+            // Except for preflight, unspecified and disconnect messages we
+            // should drop messages if all agents are blocked for the given
+            // peer URL and space id. We do not close the connection because
+            // agents in other spaces may not be blocked on the same conductor.
+            if !self.check_message_permitted(
+                &peer,
+                &space_id,
+                &module_id,
+                &message_type,
+            )? {
+                return Ok(());
```

Comment:
We need to be able to distinguish the error case from the "blocked" case since in the latter we only drop the message while in the former we close the connection, therefore, I changed the return value of `check_message_permitted` to a boolean now.